lib/Dialect/TTNN/Pipelines/TTNNPipelines.cpp

```diff
@@ -111,6 +119,7 @@ void createTTIRToTTNNBackendPipeline(
     OpPassManager &pm, const TTIRToTTNNBackendPipelineOptions &options) {
   createTTNNPipelineTTIRPasses(pm, options);
   createTTNNPipelineLoweringPasses(pm, options);
+  createTTNNPipelineWorkaroundPass(pm, options);
```

Comment:
Should this run before optimizer, or after, or it should be one or another?
Also you should add pipeline test with Optimizer and Workaround enabled.

Reply:
It should run before optimizer. In this PR I am scoping just to add a proof of concept. This pass is disabled at the moment. I will enable it with the next PR when I will cover the full set of tests.